models/Recording.js

```diff
@@ -115,9 +115,43 @@ module.exports = function(sequelize, DataTypes) {
       offset: offset,
       attributes: this.userGetAttributes,
     };
-    return this.findAndCount(q);
+
+    var queryResponse = await this.findAndCount(q);
+    filterRecordings(user, queryResponse.rows, filterOptions);
+    return queryResponse;
   };
 
+  function filterRecordings(user, recordings, options = {}) {
+    if (typeof options.latLongAcc != 'number') {
+      options.latLongAcc = 100;
+    }
+    if (!user.hasGlobalWrite()) {
+      options.latLongAcc = Math.max(options.latLongAcc, 100);
+    }
+
+    for (var i in recordings) {
+      if (recordings[i].location) {
```

Comment:
suggest you start the loop with `const rec = recordings[i];` to avoid having to index into recordings all the time.